File: jar-common/src/main/java/com/beancounter/common/model/Totals.java
package com.beancounter.common.model;


import java.math.BigDecimal;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class Totals {
  @Builder.Default
  private BigDecimal total = BigDecimal.ZERO;
}

File: jar-common/src/main/kotlin/com/beancounter/common/model/Totals.kt
package com.beancounter.common.model

import java.math.BigDecimal

/**
 * A Value, defaulting to ZERO.
 */
data class Totals(
    val currency: Currency,
    var marketValue: BigDecimal = BigDecimal.ZERO,
    var purchases: BigDecimal = BigDecimal.ZERO,
    var sales: BigDecimal = BigDecimal.ZERO,
    var cash: BigDecimal = BigDecimal.ZERO,
    var income: BigDecimal = BigDecimal.ZERO,
    var gain: BigDecimal = BigDecimal.ZERO,
)
